packages/core/src/utils/transformKeys.ts

```diff
@@ -0,0 +1,38 @@
+import { outdent } from './outdent';
+
+class IllegalFieldNameError extends Error {
+  constructor(illegalSequence: string) {
+    super();
+    this.message = outdent`
+      The sequence "${illegalSequence}" is reserved and not allowed in field names
+      Either:
+        - remove all instances of "${illegalSequence}" in the names of fields in your content
+        - add a fieldTransform function to your flatbread.config.js to translate to something else
+          Example:
+            {
+              ...,
+              fieldTransform: (value) => value.replaceAll("${illegalSequence}",'-')
+            }
+      `;
+  }
+}
+
+function isObject(obj: any): obj is Object {
+  return obj != null && obj.constructor.name === 'Object';
+}
```

Comment:
We can tighten up this predicate with:
```suggestion
function isObject<K extends string | number | symbol, T>(
  value: unknown
): value is Record<K, T> {
  return (
    !!value && typeof value === 'object' && value.constructor.name === 'Object'
  );
}
```

Reply:
Also we should move this utility to a shared `core/src/utils/objectUtils.ts`